server/instances.go

```diff
@@ -173,6 +174,13 @@ func (p *Plugin) UninstallInstance(instanceID types.ID, instanceType InstanceTyp
 			var err error
 			instance, err = p.instanceStore.LoadInstance(instanceID)
 			if err != nil {
+				if strings.Contains(err.Error(), "not found") {
+					instances.Delete(instanceID)
+					updated = instances
+					if err = p.instanceStore.StoreInstances(instances); err != nil {
+						return err
+					}
+				}
 				return err
 			}
```

Comment:
In the case where line 180 _doesn't_ have an error, I think we should return nil, instead of returning the error on the next line after the block completes. The deletion seems "successful" to me in that case.

Reply:
The recent changes don't address the above comment. We should still be returning the error at 184. Here's what I'm thinking:

```go
instance, err = p.instanceStore.LoadInstance(instanceID)
if err != nil {
    if strings.Contains(err.Error(), "not found") {
        instances.Delete(instanceID)
        updated = instances
        if err = p.instanceStore.StoreInstances(instances); err != nil {
            return err
        } else {
            return nil
        }
    }
    return err
}
```